Source/NSStoryboard.m

```diff
@@ -0,0 +1,975 @@
+{
+  // this is never encoded directly...
+}
+
+- (id) instantiate
+{
+  _storyboard = [NSStoryboard storyboardWithName: _storyboardName
+                                          bundle: [NSBundle mainBundle]];
+  return [_storyboard instantiateInitialController];
+}
+
+@end
+
+@implementation NSStoryboard
+
+- (NSXMLElement *) _createCustomObjectWithId: (NSString *)ident
+                                   userLabel: (NSString *)userLabel
+                                 customClass: (NSString *)className
+{
+  NSXMLElement *customObject =
+    [[NSXMLElement alloc] initWithName: @"customObject"];
+  NSXMLNode *idValue =
+    [NSXMLNode attributeWithName: @"id"
+                     stringValue: ident];
+  NSXMLNode *usrLabel =
+    [NSXMLNode attributeWithName: @"userLabel"
+                     stringValue: userLabel];
+  NSXMLNode *customCls =
+    [NSXMLNode attributeWithName: @"customClass"
+                     stringValue: className];
+  
+  [customObject addAttribute: idValue];
+  [customObject addAttribute: usrLabel];
+  [customObject addAttribute: customCls];
+
+  AUTORELEASE(customObject);
+  
+  return customObject;
+}
+
+- (void) _processStoryboard: (NSXMLDocument *)storyboardXml
+{
+  NSArray *docNodes = [storyboardXml nodesForXPath: @"document" error: NULL];
+
+  if ([docNodes count] > 0)
+    {
+      NSXMLElement *docNode = [docNodes objectAtIndex: 0];
+      NSArray *array = [docNode nodesForXPath: @"//scene" error: NULL];
+      NSEnumerator *en = [array objectEnumerator];
+      NSXMLElement *e = nil;  
+      NSString *customClassString = nil;
+      
+      // Set initial view controller...
+      ASSIGN(_initialViewControllerId, [[docNode attributeForName: @"initialViewController"] stringValue]);             
+      _scenesMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      _controllerMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      _documentsMap = [[NSMutableDictionary alloc] initWithCapacity: [array count]];
+      
+      while ((e = [en nextObject]) != nil)
+        {
+          NSXMLElement *doc = [[NSXMLElement alloc] initWithName: @"document"];
+          NSArray *children = [e children];
+          NSEnumerator *ce = [children objectEnumerator];
+          NSXMLElement *child = nil;
+          NSXMLDocument *document = nil;
+          NSString *sceneId = [[e attributeForName: @"sceneID"] stringValue]; 
+          NSString *controllerId = nil;
+
+          // Copy children...
+          while ((child = [ce nextObject]) != nil)
+            {
+              if ([[child name] isEqualToString: @"point"] == YES)
+                continue; // go on if it's point, we don't use that in the app...
+              
+              NSArray *subnodes = [child nodesForXPath: @"//application" error: NULL];
+              NSXMLNode *appNode = [subnodes objectAtIndex: 0];
+              if ([[appNode name] isEqualToString: @"application"] == YES)
```

Comment:
Isn't this already enforced by the XPath above?

Reply:
Fixed